### DISCUSSION TITLE
Match mainnet and Testnet transaction size limits

### DISCUSSION THREAD

**Discussion by MBeliou** [fetchai/FIPs#22] [Ideas]
Because of various proposals related to transaction size, there is currently a difference between Testnet and Mainnet regarding transaction size limits with the Testnet limits being higher than Mainnet. (6M vs 3M)

This creates a situation where smart contracts that are valid on Testnet end up being too big to be published on Mainnet, forcing unaware developers to go back to the drawing board to optimize for size, usually after a full development cycle.
We'd like to push for an increase in transaction size limits on Mainnet in order to match Testnet.

**This would mean:**

- An increase for the _max_bytes_ parameter to 600_000
- An increase for the _max_gas_ parameter to 6_000_000

Giving us the following ParameterChangeProposal changes message:
``` 
[
    {
      "subspace": "baseapp",
      "key": "BlockParams",
      "value": "{"max_bytes":"600000","max_gas":"6000000"}"
    }
 ]
```
**Some technical info:**
The 6M gas transaction size limit on Testnet was put for a vote in [september](https://fetchstation.azoyalabs.com/dorado/governance/6/) and has been live since.
The Juno network has recently put a vote to increase back max gas settings to [100M](https://www.mintscan.io/juno/proposals/274).

**Comment by Atari-buzzk1LL** [comment]
This is more than a fair proposal to put forward. Testnet and Mainnet should match in all ways unless there are issues or instabilities found in something implemented on Testnet. In this case, though, there haven't been any reported problems, so I hope to see your FIP implemented.

**Comment by crm1801** [comment]
good job!!

**Reply by cosmosscanner** [reply]
Indeed!

**Comment by Jackson-DKMG** [comment]
sounds like a good idea

**Comment by pbukva** [comment]
Resolved with governance proposal [#36.  Raise the bock limit params `max_gas` and `max_bytes` to align them with values on Dorado testnet](https://www.mintscan.io/fetchai/proposals/36), see the related [fip-006.md](https://github.com/fetchai/FIPs/blob/main/FIPs/fip-006.md).
